src/main/java/emissary/util/PayloadUtil.java

```diff
@@ -26,6 +31,38 @@ public class PayloadUtil {
     private static final String LS = System.getProperty("line.separator");
     private static final Pattern validFormRegex = Pattern.compile("^[\\w-)(/]+$");
 
+    protected static Map<String, String> historyPreference = new HashMap<>();
+
+    protected static final String REDUCED_HISTORY = "REDUCED_HISTORY";
+    protected static final String NO_URL = "NO_URL";
+
+    static {
+        configure();
+    }
+
+    protected static void configure() {
+        Configurator configG = null;
+        try {
+            configG = ConfigUtil.getConfigInfo(PayloadUtil.class);
+        } catch (IOException e) {
+            logger.error("Cannot open default config file", e);
+        }
+
+        // fill lists with data from PayloadUtil.cfg for transform itinerary/history
+        if (configG != null) {
+            for (ConfigEntry configEntry : configG.getEntries()) {
```

Comment:
I went with the "first in wins" idea because that was what was mentioned by @drivenflywheel in your last change request. Yes, REDUCED_HISTORY isn't guaranteed if it's not the first in, but there wasn't per-say one preference that carried precedence over others in my mind. If there was, I figured that would be listed first in the cfg so it would be grabbed first anyway.

I wrote out the second option you mentioned first, and then changed it to the way I ended up making it, because instead of populating the whole map and then checking if there are multiple entries, that is done when all the entries are grabbed one by one with the `config.getEntries()`. Add the first found example for a fileType, and then if it is found again, log the warning message saying XML is using whatever the first matching preference was. (As I typed this out, realized I also need to add a warning message to what I have if the same entry, i.e. `REDUCED_HISTORY = XML`, is in the same list in the cfg more than once haha)

If the concern is other non-history-itinerary things being added to cfg down the road, then yes, option two of grabbing both lists and adding them to the map, then having the multiple entries check is the better option. More code would have to be added in the future for any other history configurations, but allows for the most control as you said.

Reply:
My thinking was somewhere in the realm of @dev-mlb's two options. `config.getEntries` is too wide a swing for my comfort.  Hopefully we're overestimating the need for future-proofing; I think Option 2 is great for now, and we make it a future problem if we need even more history formats.  I don't care which way we choose to handle conflicts (first in, last in, X > Y, whatever) , so long as we document the choice right in the code itself.

Reply:
Alrighty then I will alter it to be like the second option mentioned!